src/IssueViz.Security.UnitTests/OpenInIDE/Api/OpenInIDERequestHandlerTests.cs

```diff
@@ -19,72 +19,251 @@
-                HotspotKey = "any",
-                OrganizationKey = "org"
-            };
-            var serverMock = new Mock<ISonarQubeService>();
+            InitializeStateValidator(ValidRequest, false);
 
-            IOpenInIDERequestHandler testSubject = new OpenInIDERequestHandler(validatorMock.Object, serverMock.Object, Mock.Of<ILogger>());
+            // Act
+            await testSubject.ShowHotspotAsync(ValidRequest)
+                .ConfigureAwait(false);
 
-            await testSubject.ShowHotspotAsync(request);
+            CheckInfoBarCleared();
+            CheckInfoBarShown();
+            CheckCalled(toolWindowService, stateValidatorMock);
+            CheckNotCalled(serverMock, converterMock, navigatorMock, storeMock);
+        }
+
+        [TestMethod]
+        public async Task ShowHotspot_FailedToRetrieveHotspotData_NoFurtherProcessing()
+        {
+            InitializeStateValidator(ValidRequest, true);
+            SetServerResponse(ValidRequest, null);
 
-            validatorMock.VerifyAll();
-            serverMock.Invocations.Should().BeEmpty();
+            // Act
+            await testSubject.ShowHotspotAsync(ValidRequest)
+                .ConfigureAwait(false);
+
+            CheckInfoBarCleared();
+            CheckInfoBarShown();
+            CheckCalled(toolWindowService, stateValidatorMock, serverMock);
+            CheckNotCalled(converterMock, navigatorMock, storeMock);
         }
 
-        private static Mock<IOpenInIDEStateValidator> CreateValidator(string uri, string projectKey, string orgKey, bool canHandleRequest)
+        [TestMethod]
+        public async Task ShowHotspot_ConversionFailed_NoFurtherProcessing()
         {
-            var mockValidator = new Mock<IOpenInIDEStateValidator>();
-            mockValidator.Setup(x => x.CanHandleOpenInIDERequest(It.Is<Uri>(x => x.ToString() == uri), projectKey, orgKey))
+            InitializeStateValidator(ValidRequest, true);
+            SetServerResponse(ValidRequest, ValidServerHotspot);
+
+            // Act
+            await testSubject.ShowHotspotAsync(ValidRequest)
+                .ConfigureAwait(false);
+
+            CheckInfoBarCleared();
+            CheckInfoBarShown();
+            CheckCalled(toolWindowService, stateValidatorMock, serverMock, converterMock);
+            CheckNotCalled(navigatorMock, storeMock);
+        }
+
+        [TestMethod]
+        public async Task ShowHotspot_DataIsValid_NavigationSucceeded_And_IssueAddedToStore()
+        {
+            // Note: this test needs a viz mock that doesn't have any members mocked using Setup(...).
+            // This is because we don't expect any of the members to be called (and if the mock has any
+            // members mocked then the call to converterMock.VerifyAll() will fail when the library tries
+            // to transitively verify the viz members).
+            var hotspotVizMock = new Mock<IAnalysisIssueVisualization>();
```

Comment:
I'm a bit confused by the comment.
What would happen if we use `Mock.Of<IAnalysisIssueVisualization>()` ?

Reply:
That would work; the only reason to have a mock was to check it wasn't used.